tests/rptest/services/storage.py

```diff
@@ -75,8 +77,12 @@ def add_files(self, files):
             seg.add_file(fn, ext)
 
     def set_segment_size(self, segment_name: str, size: int):
-        seg, _ = os.path.splitext(segment_name)
-        if not re.match(r"^\d+\-\d+\-v\d+$", seg):
+        """Set the data size of a segment: this is not the physical size of
+           all the segment's files, but just the size of the data part, excluding
+           space used by any indices.  This is usually what you care about, because
+           it's how Redpanda itself reasons about size for retention."""
+        seg, ext = os.path.splitext(segment_name)
+        if not (re.match(r"^\d+\-\d+\-v\d+$", seg) and ext == ".log"):
```

Comment:
Can the regex take care of both the name and extension? Eg `r"^\d+\-\d+\-v\d+\.log$"`

Reply:
Yes, this could be cleaned up to single regex instead of the splitext+if, but in this case I was doing a minimal fix